gravitee-apim-repository/gravitee-apim-repository-test/src/test/java/io/gravitee/repository/management/TagRepositoryTest.java

```diff
@@ -127,4 +127,14 @@ public void shouldFindByIdAndReference() throws Exception {
         assertEquals("Other", tag.get().getName());
         assertEquals("Description for other tag", tag.get().getDescription());
     }
+
+    @Test
+    public void should_find_by_ids_and_reference_id_and_reference_type() throws Exception {
+        final Set<Tag> tags = tagRepository.findByIdsAndReference(Set.of("international", "stores", "not-to-be-found"), "DEFAULT", TagReferenceType.ORGANIZATION);
+
+        assertNotNull(tags);
+        assertEquals(2, tags.size());
+        final Tag tagProduct = tags.stream().filter(tag -> "international".equals(tag.getId())).findAny().get();
+        assertEquals("International", tagProduct.getName());
```

Comment:
or with AssertJ

```suggestion
        assertThat(tags).hasSize(2)
        	.extract("name")
        	.contains("International");
```

(I wrote this without compiling maybe I does not work directly 🙈 )